File: backend/app-store/src/main/java/app/product/model/ProductSearchCriteria.java
package app.product.model;

import app.product.model.utill.ProductCategory;
import lombok.Builder;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
@Builder
public class ProductSearchCriteria {
    private ProductCategory productCategory;
    private String name;
    private String producer;
    private String guarantee;
    private String model;
    private String price;
    private int quantity;
}

File: frontend/app-store-ui/client/src/app/api/models/product-search-criteria.ts
/* tslint:disable */
/* eslint-disable */
import { ProductCategory } from '../models/product-category';
export interface ProductSearchCriteria {
  guarantee?: string;
  model?: string;
  name?: string;
  price?: string;
  producer?: string;
  productCategory?: ProductCategory;
  quantity?: number;
}
